### PR TITLE
[DBEX] add new 0781 mental health intro and events types pages

### PR DESCRIPTION
**Pull request by freeheeling**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
  - In the ["Add event" section](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=9856-83462&t=ymVYrSth7bWoMpuD-4) of the new, digitized Form 0781 flow, create:
    - placeholder for a new ["Mental health support" page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=10498-213662&t=NszIlX2RqUx8XIU5-4)
    - placeholder for a new ["Events intro" page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=10498-213692&t=NszIlX2RqUx8XIU5-4)
    - most of the content and functionality for a new ["Event type" page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=10498-213692&t=NszIlX2RqUx8XIU5-4)
  - page content to be implemented in [#97079](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97079) 
- _(If bug, how to reproduce)_
  - n/a
- _(What is the solution, why is this the solution)_
  - Create a new flow and pages to align the digital form with the [VA Form 21-0781](https://www.vba.va.gov/pubs/forms/VBA-21-0781-ARE.pdf) paper version 
- _(Which team do you work for, does your team own the maintenance of this component?)_
  - Disability Benefits Experience Team 2 (dBeX Carbs 🥖), which owns maintenance of all files in this PR
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
  - `sync_modern_0781_flow` will be utilized for specific users until all who start a new form and claim a new disability will be routed through an alternative 0781 flow

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/97078

## Testing done

- _Describe what the old behavior was prior to the change_
 - This page did not previously exist on the digital form
- _Describe the steps required to verify your changes are working as expected_
  1. in the development environment, ensure that the [`sync_modern_0781_flow`](https://dev-api.va.gov/flipper/features#:~:text=disability_compensation_sync_modern_0781_flow) toggle is enabled 
  2. start a new claim
  3. claim at least 1 new condition
  4. select at least 1 of the new conditions on the [0781 screener page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=9250-79794&t=NszIlX2RqUx8XIU5-4)
  5. once https://github.com/department-of-veterans-affairs/va.gov-team/issues/97067 has been completed, the 3 new pages should display after the new [0781 choice page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=9442-72041&t=NszIlX2RqUx8XIU5-4), only if the option to complete the online form is selected on the preceding choice page
  6. the last of the new pages should continue to the "Supporting evidence" chapter, until additional pages are developed
- Describe the tests completed and the results
  - added unit tests
  - manually tested

## Screenshots

| Page | Mobile view (with URL) |
| ----- | ------- |
| Mental health support | Screenshot 2024-12-20 at 11 52 21 AM |
| Events intro | Screenshot 2024-12-20 at 12 05 52 PM |
| Event type | Screenshot 2024-12-20 at 12 04 42 PM |

## What areas of the site does it impact?

- 526 EZ (within a new "Additional Forms" chapter)

### Quality Assurance & Testing

- [x] I ~fixed~|~updated~|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) ~has been performed~ **(linked page does not exist)**

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/utils/form0781.js [hunk 1]
```diff
@@ -23,3 +24,19 @@ export function showForm0781Pages(formData) {
...
+ * 2. the option to complete the online form is selected
+ *
+ * @param {object} formData
+ * @returns {boolean} true if COMPLETE_ONLINE_FORM is selected, false otherwise
+ */
+export function isCompletingForm0781(formData) {
+  return (
+    showForm0781Pages(formData) &&
+    formData['view:mentalHealthWorkflowChoice'] ===
+      form0781WorkflowChoices.COMPLETE_ONLINE_FORM
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/utils/form0781.js:40]: @NB28VT, I based this off of your changes in [dc27a78](https://github.com/department-of-veterans-affairs/vets-website/pull/33655/commits/dc27a78ca4943e594109738658624a480ba67397)

File: src/applications/disability-benefits/all-claims/constants.js [hunk 2]
```diff
@@ -403,3 +403,11 @@ export const ADDITIONAL_EXPOSURES = Object.freeze({
   none: 'None of these',
   notsure: 'I’m not sure if I have been exposed to these hazards',
 });
+
```
Comment (ajones446) [inline src/applications/disability-benefits/all-claims/constants.js:406]: We may want to decide on a different way to organize constants as we keep adding content - like a new file for `/all-claims/constants/form0781` - but this is good for now.

**Review by ajones446 — APPROVED** [review]
LGTM! I tried it locally and was able to click through and see the data change (after forcing `isCompletingForm0781(formData)` to return true to get the pages to render).

File: src/applications/disability-benefits/all-claims/constants.js (hunk 2 above)
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/constants.js:406]: Valid suggestion.

**Comment by freeheeling** [comment]
The [failing unit test](https://github.com/department-of-veterans-affairs/vets-website/actions/runs/12507360142/job/34893978721?pr=33741) is **NOT** attributed to changes in this PR.

**Review by IGallupSoCo — APPROVED** [review]
LGTM

**Pull request merged**